### DISCUSSION TITLE
Multiple remote controllers connect to multiple SPIKE Hub

### DISCUSSION THREAD

**Discussion by ongpingyang** [pybricks/support#1266] [Q&A]
Hi, I have recently tried a Powered Up remote controller to connect to a SPIKE hub. It works on one-to-one connection, but I wish to use multiple controllers to control different hub at the same time, for something like remote control SPIKE robot match.

From my testing, the controller will connect to the first hub which run the program. If I run the program of two hubs at the same time, the remote seems to randomly pick and connect to one of the brick.

The example I refer to is https://pybricks.com/projects/tutorials/wireless/remote-control/drive-base/ and it only shows "rc = Remote()".

Is there anyway to set a controller to a specific hub? Or is there any documentation on this part?

Thanks :D

**Comment by BertLindeman** [comment]
Hello,

You might take a look at #1259 LEGO 42146 Liebherr LR 1300 crane with one remote connected to each Technic hub
The main program loads the specific program for that hub (depending on it's name).
The hub program connects to a remote and verifies the name of it.

Maybe this helps a bit.

Bert

**Reply by ongpingyang** [reply]
Great project! I can use the idea to control more of technic set :D

But one more question, how do you download/change controller name?

Is it that you download the code to hub (eg tophub.py), connect the controller to the hub then it will change name because of this line of code -- "REMOTENAMES = lego_42146_liebherr_config.REMOTENAMES"?

**Reply by BertLindeman** [reply]
Thank you, 😄 

In the zip file over at bricksafe [bricksafe](https://bricksafe.com/files/Bert_bricksafe/42146-lr-13000/lego_42146_pybricks.zip) there is a small program `lego_42146_set_remote_names.py` that can set the remote controller names.
That module also imports the config module, so the correct name is set.
Added the source below.

<details>
<summary> click to open lego_42146_set_remote_names.py</summary>

```python

""" Module function:
        Utility to set the names of the two remotes for the pybricks version of LEGO 42146 Liebherr LR 13000.
        Set the remote names as set in lego_42146_liebherr_config.py
        Run this program separately for each of the two remotes.
        So run twice; once for each remote.

    Load all lego_42146_*.py programs into beta.pybricks.com on Google Chrome or another compatible browser.
    After that:
        Run THIS program twice:
         - once to set the name of the left remote.
         - once to set the name of the right remote.
"""

# from pybricks.hubs import TechnicHub
from pybricks.pupdevices import Remote
from pybricks.tools import wait

# Import required MicroPython libraries.
from usys import stdin
from uselect import poll

import lego_42146_liebherr_config

# use the remote names as specified in the config module.
REMOTENAMES = lego_42146_liebherr_config.REMOTENAMES


def set_remote(_side, new_name, _remote):
    """Procedure to set the name of one remote"""
    print("Press center (green) button on", _side, "REMOTE.")
    wait(2000)  # best have a press on the remote before the hub starts searching
    print("Waiting 20 seconds for a remote connection to remote.")
    _remote = Remote(timeout=20000)

    print("Connected to remote\t", _remote.name())

    # set new name
    _remote.name(new_name)
    print("Remote name set to \t", _remote.name())
    print("\n\t\tDone")


REMOTE = None

print()
# set a remote-name.
print("Which remote to set the name?")
print("\n\tL or l for the left remote")
print("\tR or r for the right remote")
print("\n\tClick on the log panel (HERE) and type a letter")

# Register the standard input so we can read keyboard presses.
keyboard = poll()
keyboard.register(stdin)

while True:
    # Check if a key has been pressed.
    if keyboard.poll(0):

        # Read the key and print it.
        key = stdin.read(1)
        if key in ['L', 'L'.lower()]:
            print('You chose the left remote...')
            set_remote("left", REMOTENAMES["left"], REMOTE)
            break

        if key in ['R', 'r']:
            print('You chose the right remote...')
            set_remote("right", REMOTENAMES["right"], REMOTE)
            break

        print("Got key '" + str(key) + "' not a valid choice")

```

</details>

Bert

**Comment by laurensvalk** [comment]
>  I wish to use multiple controllers to control different hub at the same time

This is also possible. You can give each remote a unique Bluetooth name, and then make each hub search only for that particular remote. Then the connectivity is like this:

```
crane_remote   -----> crane_hub
tank_remote   -----> tank_hub
```

Another approach is like this:

```
remote ----> crane_hub --------------------> tank_hub
                         (via broadcast)
```

**Reply by ongpingyang** [reply]
Thanks for the suggestion!

Is there any extra documentation on Powered Up remote? Because I can't find the command like setting remote name in the tutorial/documentation 😅

I'm new to Pybricks and need some help in referring modules and parameters 😄

**Reply by laurensvalk** [reply]
You could use the documentation in the right-hand side panel or view it [online](https://docs.pybricks.com/en/latest/pupdevices/remote.html).

There's examples for renaming the remote too.
